### PR TITLE
Cleaned up TimeWindow and added a bit more docs.

### PR DESCRIPTION
**Pull request by shamsasari**
I also made `midpoint` return null rather than throwing an exception if either side is null.

### CODE REVIEW

File: core/src/test/kotlin/net/corda/core/node/services/TimeWindowCheckerTests.kt [hunk 1]
```diff
@@ -5,12 +5,12 @@ import net.corda.core.utilities.seconds
...
 import java.time.Clock
 import java.time.Instant
-import java.time.ZoneId
+import java.time.ZoneOffset
 import kotlin.test.assertFalse
 import kotlin.test.assertTrue
 
 class TimeWindowCheckerTests {
-    val clock = Clock.fixed(Instant.now(), ZoneId.systemDefault())
+    val clock: Clock = Clock.fixed(Instant.now(), ZoneOffset.UTC)
```
Comment (adagys) [inline core/src/test/kotlin/net/corda/core/node/services/TimeWindowCheckerTests.kt]: what's the difference here?

Comment (shamsasari) [inline core/src/test/kotlin/net/corda/core/node/services/TimeWindowCheckerTests.kt]: Make the tests deterministic I guess, across locations. Not sure that it matters, but not harm either.

File: core/src/main/kotlin/net/corda/core/contracts/TimeWindow.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
+package net.corda.core.contracts
+
+import net.corda.core.internal.div
+import net.corda.core.internal.until
+import net.corda.core.serialization.CordaSerializable
+import java.time.Duration
+import java.time.Instant
+
+/**
+ * A time-window is required for validation/notarization purposes. If present in a transaction, contains a time that was
```
Comment (gendal) [inline core/src/main/kotlin/net/corda/core/contracts/TimeWindow.kt]: is it true that it is required??  Contradicts the "if present" statement in the next sentence

Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/contracts/TimeWindow.kt]: That is true. Looks like i didn't read this existing paragraph properly. I'll submit a PR to fix.

**Pull request merged**